server/src/main/java/org/elasticsearch/index/query/ExistsQueryBuilder.java

```diff
@@ -165,7 +167,7 @@ private static Query newFieldExistsQuery(QueryShardContext context, String field
             if (context.getObjectMapper(field) != null) {
                 return newObjectFieldExistsQuery(context, field);
             }
-            return Queries.newMatchNoDocsQuery("No field \"" + field + "\" exists in mappings.");
+            return Queries.newMatchNoDocsQuery("User requested \"match_none\" query.");
```

Comment:
this line is unreachable right? maybe throw an IllegalStateException instead? If I'm wrong and this is not dead code, would you mind explaining the conditions under which it is triggered?

Reply:
No, when User looks for an Object field that does not exist in the mapping, it will fall into this.